File: PDOO/p4/civitas_rb/lib/casilla_juez.rb
module Civitas

class Casilla_juez < Casilla
  
  def initialize(numCasillaCarcel, nombre)
    super(nombre)
    @carcel=numCasillaCarcel
  end
  
  def recibejugador(iactual, todos)
    if(jugador_correcto(iactual, todos))
      informe(iactual, todos)
      todos.at(iactual).encarcelar(@carcel)
    end
  end
  
  def to_string()
    return "#{super}, { carcel= #{@carcel} }"
  end
  
end
end

File: PDOO/p4/Civitas/src/civitas/CasillaJuez.java

package civitas;

import java.util.ArrayList;

public class CasillaJuez extends Casilla {
    private static int carcel;
    
    CasillaJuez(int numCasillaCarcel, String nombre){
        //Para crear la casilla de tipo juez
        super(nombre);
        carcel = numCasillaCarcel;
    }
    
    @Override
    public void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if (super.jugadorCorrecto(iactual, todos)){
            super.informe(iactual, todos);
            todos.get(iactual).encarcelar(carcel);
        }
    }
    
    @Override
    public String toString(){
        return "Casilla{ nombre=" + super.getNombre() + ", carcel=" + carcel + " }";
    }
}
